lib/reconcile/utils.go

```diff
@@ -378,6 +378,83 @@ func getAstarteCommonEnvVars(deploymentName string, cr *apiv1alpha1.Astarte, bac
 	return appendRabbitMQConnectionEnvVars(ret, rpcPrefix+"RPC_AMQP_CONNECTION", cr)
 }
 
+func appendCassandraConnectionEnvVars(ret []v1.EnvVar, prefix string, cr *apiv1alpha1.Astarte) []v1.EnvVar {
+	spec := cr.Spec.Cassandra.Connection
+	if spec != nil {
+		// pool size
+		if spec.PoolSize != "" {
+			ret = append(ret,
+				v1.EnvVar{
+					Name:  prefix + "_POOL_SIZE",
+					Value: strconv.Itoa(int(spec.PoolSize)),
+				},
+			)
+		}
+
+		// autodiscovery
+		if pointy.BoolValue(spec.AutodiscoveryEnabled, true) {
+			ret = append(ret,
+				v1.EnvVar{
+					Name:  prefix + "_AUTODISCOVERY_ENABLED",
+					Value: "true",
+				},
+			)
+		}
+
+		// SSL
+		if spec.SSLConfiguration.Enabled {
+			ret = append(ret, v1.EnvVar{
+				Name:  prefix + "_SSL_ENABLED",
+				Value: "true",
+			})
+
+			// CA configuration
+			if spec.SSLConfiguration.CustomCASecret.Name != "" {
+				// getAstarteCommonVolumes will mount the volume for us, if we're here. So trust the rest of our code.
+				ret = append(ret, v1.EnvVar{
+					Name:  prefix + "_SSL_CA_FILE",
+					Value: "/cassandra-ssl/ca.crt",
+				})
+			}
+
+			// SNI configuration
+			switch {
+			case spec.SSLConfiguration.CustomSNI != "":
+				ret = append(ret, v1.EnvVar{
+					Name:  prefix + "_SSL_CUSTOM_SNI",
+					Value: spec.SSLConfiguration.CustomSNI,
+				})
+			case !pointy.BoolValue(spec.SSLConfiguration.SNI, true):
+				ret = append(ret, v1.EnvVar{
+					Name:  prefix + "_SSL_DISABLE_SNI",
+					Value: "true",
+				})
+			}
+		}
+	}
+
+	// Fetch our Credentials for Cassandra
```

Comment:
This part is wrong I think.  It might as well be possible that there's no authentication requested. In that case, you shouldn't set those vars.